### PR TITLE
Migrate SQFormDateTimePicker to Typescript

### PR DESCRIPTION
**Pull request by hvilander**

### CODE REVIEW

File: src/components/SQForm/SQFormDateTimePicker.tsx [hunk 1]
```diff
@@ -1,19 +1,37 @@
...
+  /** Disabled property to disable the input if true */
+  isDisabled?: boolean;
+  /** Required property used to highlight input and label if not fulfilled */
+  isRequired?: boolean;
+  /** Placeholder text used inside the input field to provide hints to the user */
+  placeholder?: string;
+  /** Custom onBlur event callback */
+  onBlur?: React.FocusEventHandler<HTMLInputElement>;
+  /** Custom onChange event callback */
+  onChange?: React.ChangeEventHandler;
```
Comment (hvilander) [inline src/components/SQForm/SQFormDateTimePicker.tsx:21]: This seems to work, but I don't believe it is the correct type.

File: src/components/SQForm/SQFormDateTimePicker.tsx [hunk 2]
```diff
@@ -1,19 +1,37 @@
 import React from 'react';
-import PropTypes from 'prop-types';
+import moment from '@material-ui/pickers/adapter/moment';
```
Comment (codyarose) [inline src/components/SQForm/SQFormDateTimePicker.tsx:2]: Is this the same thing as `import moment from 'moment'`?

Comment (hvilander) [reply src/components/SQForm/SQFormDateTimePicker.tsx:2]: It is not the same, as the adapter is not the same as the moment library itself. MUI supplies the adapter I believe for typing. I want to find some docs to be more sure.

Comment (hvilander) [reply src/components/SQForm/SQFormDateTimePicker.tsx:2]: It looks like I am off base. I am not sure where I pulled my original line from. I will update it.

**Comment by hvilander** [comment]
Please hold off on merging this. When I came back to this after letting it sit for awhile I am seeing a bug somewhere in validation.

File: src/components/SQForm/SQFormDateTimePicker.tsx (hunk 1 above)
Comment (hvilander) [reply src/components/SQForm/SQFormDateTimePicker.tsx:21]: The SQFormDatePicker did it differently and seems to work fine. I am going to align with that.

**Review by 20BBrown14 — APPROVED** [review]

File: src/components/SQForm/SQFormDateTimePicker.tsx [hunk 3]
```diff
@@ -1,10 +1,27 @@
...
 import {ClickAwayListener, makeStyles} from '@material-ui/core';
 import {useForm} from './useForm';
+import BaseFieldProps from '../../types/BaseFieldProps';
+import {ParsableDate} from '@material-ui/pickers/constants/prop-types';
+
+interface SQFormDateTimePickerProps extends BaseFieldProps {
+  /** Disabled property to disable the input if true */
+  isDisabled?: boolean;
+  /** Required property used to highlight input and label if not fulfilled */
+  isRequired?: boolean;
```
Comment (20BBrown14) [inline src/components/SQForm/SQFormDateTimePicker.tsx:15]: I think we've gone away with this prop on our components since it should be automatic now

Comment (hvilander) [reply src/components/SQForm/SQFormDateTimePicker.tsx]: quite right, removing it.

**Pull request merged**